javascript/script.js

```diff
@@ -79,28 +81,42 @@ function handleUrlFragment(searchInput) {
 }
 function filterWebring(searchTerm) {
   const searchLower = searchTerm.toLowerCase();
-  const filteredSites = webringData.sites.filter(
-    (site) =>
+  const matchedSiteIndices = [];
+  webringData.sites.forEach((site, index) => {
+    if (
       site.name.toLowerCase().includes(searchLower) ||
-      site.website.toLowerCase().includes(searchLower) ||
+      fuzzyMatch(site.website.toLowerCase(), searchLower) ||
       site.year.toString().includes(searchLower)
-  );
-  createWebringList(filteredSites);
+    ) {
+      matchedSiteIndices.push(index);
+    }
+  });
+  createWebringList(matchedSiteIndices);
 }
 let navigateWebring = () => {
   // https://cs.uwatering.com/#your-site-here?nav=next OR
   // https://cs.uwatering.com/#your-site-here?nav=prev
   const fragment = window.location.hash.slice(1); // #your-site-here?nav=
   if (!fragment.includes("?")) return;
+
   const [currentSite, query] = fragment.split("?");
   const params = new URLSearchParams(query);
   const nav = params.get("nav");
   if (!nav || !["next", "prev"].includes(nav)) return;
 
+  const match = webringData.sites.filter((site) =>
+    fuzzyMatch(currentSite, site.website)
```

Comment:
theres two ways we could do this comparison with the current fuzzyMatch function:
1. check if `currentSite` contains `site.website`
2. check if `site.website` contains `current.site`

Using ivan's website for comparison, 2. would let urls like https://iva slip through (`ivan-yu.ca.contains(iva) === true`) among other false positives like empty strings. 

approach would still let urls like ivan-yu.caIAmAGirraffe pass (ivan-yu.caIAmAGiraffe.contains(`ivan-yu.ca === true`) but i think this edge case is much rarer?